linker/shared/src/test/scala/org/scalajs/linker/AnalyzerTest.scala

```diff
@@ -560,6 +560,30 @@ class AnalyzerTest {
     }
   }
 
+  @Test
+  def importMetaWithoutESModule(): AsyncResult = await {
+    val classDefs = Seq(
+      classDef("A",
+        kind = ClassKind.ModuleClass, superClass = Some(ObjectClass),
+        memberDefs = List(
+          trivialCtor("A"),
+          mainMethodDef(JSImportMeta())
+        )
+      )
+    )
+
+    val moduleInitializer = ModuleInitializer.mainMethodWithArgs("A", "main")
+
+    testScriptAndModule(classDefs, moduleInitializers = List(moduleInitializer)) { scriptAnalysis =>
+      assertContainsError("ImportMetaWithoutESModule", scriptAnalysis) {
+        case ImportMetaWithoutESModule(_) =>
+          true
+      }
+    } { moduleAnalysis =>
+      assertNoError(moduleAnalysis)
+    }
```

Comment:
Hum, I do not understand why this test passes: Shouldn't it fail with `moduleKind = CommonJSModule`?

Reply:
That is weird, indeed. 🤔 I will investigate on Monday, when I have a computer again. I only have a mobile for the weekend.

Reply:
It turns out `testScriptAndModule` had a pretty several flaw. I have added a first commit that fixes that flaw, and then amended the rest to make it work.

Updated.